subprojects/core/src/integTest/groovy/org/gradle/NameValidationIntegrationTest.groovy

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+
+import static org.gradle.internal.os.OperatingSystem.WINDOWS
+import static org.gradle.internal.os.OperatingSystem.current
+
+class NameValidationIntegrationTest extends AbstractIntegrationSpec {
+
+    def "project names should not contain forbidden characters"() {
+        when:
```

Comment:
Yeah... I so much wanted `expectDeprecationWarning()` to be in the `then:` section. Which does not work with the way the fixture works. So I messed this up. I restructured the test.